`test_handler_metrics_saver_dist` error

**Describe the bug**
a frequent error from the premerge tests, e.g. 
- https://github.com/Project-MONAI/MONAI/runs/4753115139?check_suite_focus=true
- https://github.com/Project-MONAI/MONAI/runs/4773096722?check_suite_focus=true
```
tests/test_handler_metrics_saver_dist.py
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/__w/MONAI/MONAI/tests/test_handler_metrics_saver_dist.py", line 87, in _run
    for i, row in enumerate(f_csv):
OSError: [Errno 9] Bad file descriptor

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/opt/conda/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/__w/MONAI/MONAI/tests/utils.py", line 396, in run_process
    raise e
  File "/__w/MONAI/MONAI/tests/utils.py", line 387, in run_process
    func(*args, **kwargs)
  File "/__w/MONAI/MONAI/tests/utils.py", line 557, in _call_original_func
    return f(*args, **kwargs)
  File "/__w/MONAI/MONAI/tests/test_handler_metrics_saver_dist.py", line 31, in test_content
    self._run(tempdir)
  File "/__w/MONAI/MONAI/tests/test_handler_metrics_saver_dist.py", line 87, in _run
    for i, row in enumerate(f_csv):
OSError: [Errno 9] Bad file descriptor
F
======================================================================
FAIL: test_content (__main__.DistributedMetricsSaver)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/MONAI/MONAI/tests/utils.py", line 432, in _wrapper
    assert results.get(), "Distributed call failed."
AssertionError: Distributed call failed.

----------------------------------------------------------------------
Ran 1 test in 10.636s
```

--- a/tests/test_handler_metrics_saver_dist.py
+++ b/tests/test_handler_metrics_saver_dist.py
@@ -31,6 +31,7 @@ class DistributedMetricsSaver(DistTestCase):
             self._run(tempdir)
 
     def _run(self, tempdir):
+        my_rank = dist.get_rank()
         fnames = ["aaa" * 300, "bbb" * 301, "ccc" * 302]
 
         metrics_saver = MetricsSaver(
@@ -46,7 +47,7 @@ class DistributedMetricsSaver(DistTestCase):
 
         engine = Engine(_val_func)
 
-        if dist.get_rank() == 0:
+        if my_rank == 0:
             data = [{"image_meta_dict": {"filename_or_obj": [fnames[0]]}}]
 
             @engine.on(Events.EPOCH_COMPLETED)
@@ -54,7 +55,7 @@ class DistributedMetricsSaver(DistTestCase):
                 engine.state.metrics = {"metric1": 1, "metric2": 2}
                 engine.state.metric_details = {"metric3": torch.tensor([[1, 2]]), "metric4": torch.tensor([[5, 6]])}
 
-        if dist.get_rank() == 1:
+        if my_rank == 1:
             # different ranks have different data length
             data = [
                 {"image_meta_dict": {"filename_or_obj": [fnames[1]]}},
@@ -79,7 +80,7 @@ class DistributedMetricsSaver(DistTestCase):
         metrics_saver.attach(engine)
         engine.run(data, max_epochs=1)
 
-        if dist.get_rank() == 0:
+        if my_rank == 0:
             # check the metrics.csv and content
             self.assertTrue(os.path.exists(os.path.join(tempdir, "metrics.csv")))
             with open(os.path.join(tempdir, "metrics.csv")) as f:

--- a/tests/test_integration_determinism.py
+++ b/tests/test_integration_determinism.py
@@ -75,7 +75,7 @@ class TestDeterminism(DistTestCase):
     def tearDown(self):
         set_determinism(seed=None)
 
-    @TimedCall(seconds=150)
+    @TimedCall(seconds=150, skip_timing=not torch.cuda.is_available())
     def test_training(self):
         set_determinism(seed=0)
         loss, step = run_test(device=self.device)

--- a/tests/test_integration_stn.py
+++ b/tests/test_integration_stn.py
@@ -103,7 +103,7 @@ class TestSpatialTransformerCore(DistTestCase):
     def tearDown(self):
         set_determinism(seed=None)
 
-    @TimedCall(seconds=100)
+    @TimedCall(seconds=100, skip_timing=not torch.cuda.is_available())
     def test_training(self):
         """
         check that the quality AffineTransform backpropagation

--- a/tests/test_savitzky_golay_filter.py
+++ b/tests/test_savitzky_golay_filter.py
@@ -25,7 +25,7 @@ TEST_CASE_SINGLE_VALUE = [
     torch.Tensor([1.0]).unsqueeze(0).unsqueeze(0),  # Input data: Single value
     torch.Tensor([1 / 3]).unsqueeze(0).unsqueeze(0),  # Expected output: With a window length of 3 and polyorder 1
     # output should be equal to mean of 0, 1 and 0 = 1/3 (because input will be zero-padded and a linear fit performed)
-    1e-15,  # absolute tolerance
+    1e-6,  # absolute tolerance
 ]
 
 TEST_CASE_1D = [
@@ -35,21 +35,21 @@ TEST_CASE_1D = [
     .unsqueeze(0)
     .unsqueeze(0),  # Expected output: zero padded, so linear interpolation
     # over length-3 windows will result in output of [2/3, 1, 2/3].
-    1e-15,  # absolute tolerance
+    1e-6,  # absolute tolerance
 ]
 
 TEST_CASE_2D_AXIS_2 = [
     {"window_length": 3, "order": 1},  # along default axis (2, first spatial dim)
     torch.ones((3, 2)).unsqueeze(0).unsqueeze(0),
     torch.Tensor([[2 / 3, 2 / 3], [1.0, 1.0], [2 / 3, 2 / 3]]).unsqueeze(0).unsqueeze(0),
-    1e-15,  # absolute tolerance
+    1e-6,  # absolute tolerance
 ]
 
 TEST_CASE_2D_AXIS_3 = [
     {"window_length": 3, "order": 1, "axis": 3},  # along axis 3 (second spatial dim)
     torch.ones((2, 3)).unsqueeze(0).unsqueeze(0),
     torch.Tensor([[2 / 3, 1.0, 2 / 3], [2 / 3, 1.0, 2 / 3]]).unsqueeze(0).unsqueeze(0),
-    1e-15,  # absolute tolerance
+    1e-6,  # absolute tolerance
 ]
 
 # Replicated-padding trivial tests
@@ -59,7 +59,7 @@ TEST_CASE_SINGLE_VALUE_REP = [
     torch.Tensor([1.0]).unsqueeze(0).unsqueeze(0),  # Input data: Single value
     torch.Tensor([1.0]).unsqueeze(0).unsqueeze(0),  # Expected output: With a window length of 3 and polyorder 1
     # output will be equal to mean of [1, 1, 1] = 1 (input will be nearest-neighbour-padded and a linear fit performed)
-    1e-15,  # absolute tolerance
+    1e-6,  # absolute tolerance
 ]
 
 TEST_CASE_1D_REP = [
@@ -67,21 +67,21 @@ TEST_CASE_1D_REP = [
     torch.Tensor([1.0, 1.0, 1.0]).unsqueeze(0).unsqueeze(0),  # Input data
     torch.Tensor([1.0, 1.0, 1.0]).unsqueeze(0).unsqueeze(0),  # Expected output: zero padded, so linear interpolation
     # over length-3 windows will result in output of [2/3, 1, 2/3].
-    1e-15,  # absolute tolerance
+    1e-6,  # absolute tolerance
 ]
 
 TEST_CASE_2D_AXIS_2_REP = [
     {"window_length": 3, "order": 1, "mode": "replicate"},  # along default axis (2, first spatial dim)
     torch.ones((3, 2)).unsqueeze(0).unsqueeze(0),
     torch.Tensor([[1.0, 1.0], [1.0, 1.0], [1.0, 1.0]]).unsqueeze(0).unsqueeze(0),
-    1e-15,  # absolute tolerance
+    1e-6,  # absolute tolerance
 ]
 
 TEST_CASE_2D_AXIS_3_REP = [
     {"window_length": 3, "order": 1, "axis": 3, "mode": "replicate"},  # along axis 3 (second spatial dim)
     torch.ones((2, 3)).unsqueeze(0).unsqueeze(0),
     torch.Tensor([[1.0, 1.0, 1.0], [1.0, 1.0, 1.0]]).unsqueeze(0).unsqueeze(0),
-    1e-15,  # absolute tolerance
+    1e-6,  # absolute tolerance
 ]
 
 # Sine smoothing
@@ -101,18 +101,18 @@ class TestSavitzkyGolayCPU(unittest.TestCase):
     @parameterized.expand(
         [TEST_CASE_SINGLE_VALUE, TEST_CASE_1D, TEST_CASE_2D_AXIS_2, TEST_CASE_2D_AXIS_3, TEST_CASE_SINE_SMOOTH]
     )
-    def test_value(self, arguments, image, expected_data, atol):
+    def test_value(self, arguments, image, expected_data, atol, rtol=1e-5):
         result = SavitzkyGolayFilter(**arguments)(image)
-        np.testing.assert_allclose(result, expected_data, atol=atol)
+        np.testing.assert_allclose(result, expected_data, atol=atol, rtol=rtol)
 
 
 class TestSavitzkyGolayCPUREP(unittest.TestCase):
     @parameterized.expand(
         [TEST_CASE_SINGLE_VALUE_REP, TEST_CASE_1D_REP, TEST_CASE_2D_AXIS_2_REP, TEST_CASE_2D_AXIS_3_REP]
     )
-    def test_value(self, arguments, image, expected_data, atol):
+    def test_value(self, arguments, image, expected_data, atol, rtol=1e-5):
         result = SavitzkyGolayFilter(**arguments)(image)
-        np.testing.assert_allclose(result, expected_data, atol=atol)
+        np.testing.assert_allclose(result, expected_data, atol=atol, rtol=rtol)
 
 
 @skip_if_no_cuda
@@ -120,9 +120,9 @@ class TestSavitzkyGolayGPU(unittest.TestCase):
     @parameterized.expand(
         [TEST_CASE_SINGLE_VALUE, TEST_CASE_1D, TEST_CASE_2D_AXIS_2, TEST_CASE_2D_AXIS_3, TEST_CASE_SINE_SMOOTH]
     )
-    def test_value(self, arguments, image, expected_data, atol):
+    def test_value(self, arguments, image, expected_data, atol, rtol=1e-5):
         result = SavitzkyGolayFilter(**arguments)(image.to(device="cuda"))
-        np.testing.assert_allclose(result.cpu(), expected_data, atol=atol)
+        np.testing.assert_allclose(result.cpu(), expected_data, atol=atol, rtol=rtol)
 
 
 @skip_if_no_cuda
@@ -130,9 +130,9 @@ class TestSavitzkyGolayGPUREP(unittest.TestCase):
     @parameterized.expand(
         [TEST_CASE_SINGLE_VALUE_REP, TEST_CASE_1D_REP, TEST_CASE_2D_AXIS_2_REP, TEST_CASE_2D_AXIS_3_REP]
     )
-    def test_value(self, arguments, image, expected_data, atol):
+    def test_value(self, arguments, image, expected_data, atol, rtol=1e-5):
         result = SavitzkyGolayFilter(**arguments)(image.to(device="cuda"))
-        np.testing.assert_allclose(result.cpu(), expected_data, atol=atol)
+        np.testing.assert_allclose(result.cpu(), expected_data, atol=atol, rtol=rtol)
 
 
 if __name__ == "__main__":

--- a/tests/utils.py
+++ b/tests/utils.py
@@ -24,7 +24,7 @@ import warnings
 from functools import partial
 from subprocess import PIPE, Popen
 from typing import Callable, Optional, Tuple
-from urllib.error import HTTPError, URLError
+from urllib.error import ContentTooShortError, HTTPError, URLError
 
 import numpy as np
 import torch

